Spring-server/src/main/java/ikpi63holding/postclient/ApiController.java

```diff
@@ -122,4 +124,27 @@ void deleteUser(@PathVariable Long id) {
         }
     }
 
+    @PostMapping("/registration")
+    @ResponseStatus(HttpStatus.CREATED)
+    User registration(@RequestBody User newUser) {
+        PasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        
+        if (this.repository.findByUsername(newUser.getUsername()) != null) {
+            throw new ResponseStatusException(HttpStatus.ACCEPTED, "PAYLOAD");
+        }
+
+        return repository.save(new User(
+            newUser.getUsername(),
+            passwordEncoder.encode(newUser.getPassword()),
```

Comment:
What happends when password is null? Maybe we should check for it implicitly.

Reply:
If password is null we'll get NullPointerException but i think we should check it before sending of the registration request